### PR TITLE
[stubgenc] Generate Inline generics from `__type_params__`

### LINKED ISSUE #17447: [stubgen] infer_sig_from_docstring fails with python3.12 inline generic
**Bug Report**

Trying to infer types from docstrings with python3.12 inline generics fails to generate.

**To Reproduce**

```python
from mypy.stubdoc import infer_sig_from_docstring
infer_sig_from_docstring('\nfunc[T, V](x: T, y: V) -> T', 'func')
```

**Expected Behavior**

```python
infer_sig_from_docstring('\nfunc[T, V](x: T, y: V) -> T', 'func') ==
            [
                FunctionSig(
                    name="func", args=[ArgSig(name="x", type="T"), ArgSig(name="y", type="V")], ret_type="T"
                )
            ]
```

**Actual Behavior**

```python
infer_sig_from_docstring('\nfunc[T, V](x: T, y: V) -> T', 'func') == []
```

**Your Environment**

- Mypy version used: 1.11.dev
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: python 3.10

### PR DESCRIPTION
**Pull request by InvincibleRMC**
Fixes #17447 

This allows generating inline generics for stubgenc by using `__type_params__`.

Similar to #17463 but for `__type_params__` instead of `__annotations__`

### CODE REVIEW

File: mypy/test/teststubgen.py [hunk 1]
```diff
@@ -906,6 +931,50 @@ class TestClass(argparse.Action):
         assert_equal(output, ["class C(argparse.Action): ..."])
         assert_equal(gen.get_imports().splitlines(), ["import argparse"])
 
+    @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
+    def test_inline_generic_class(self) -> None:
+        T = TypeVar("T")
+
+        class TestClass:
```
Comment (sobolevn) [inline mypy/test/teststubgen.py:938]: I would also suggest to add a test case with `class Some[T]` test hidden behind `exec()` (for correct syntax) and `skipIf`.

Comment (InvincibleRMC) [reply mypy/test/teststubgen.py:938]: Do you mean something like the following?

```python
 @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
    def test_generic_class(self) -> None:

        class TestClass[T]:
            pass
        output: list[str] = []
        mod = ModuleType("module", "")
        gen = InspectionStubGenerator(mod.__name__, known_modules=[mod.__name__], module=mod)
        gen.generate_class_stub("C", TestClass, output)
        assert_equal(output, ["class C[T]: ..."])
        ```

Comment (InvincibleRMC) [reply mypy/test/teststubgen.py:938]: Follow up question should this test still live in stubgenc or should it move elsewhere to a different stubgen section?

Comment (sobolevn) [reply mypy/test/teststubgen.py:938]: Something like:

```python
    @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
    def test_generic_class(self) -> None:
        exec("class Test[A]: ...")
        ...
```

Comment (InvincibleRMC) [reply mypy/test/teststubgen.py:938]: This should now be address by the new commit

File: mypy/test/teststubgen.py [hunk 2]
```diff
@@ -906,6 +931,54 @@ class TestClass(argparse.Action):
...
+
+        output: list[str] = []
+        mod = ModuleType("module", "")
+        gen = InspectionStubGenerator(mod.__name__, known_modules=[mod.__name__], module=mod)
+        gen.generate_class_stub("C", TestClass, output)
+        assert_equal(output, ["class C[T]: ..."])
+
+    @unittest.skipIf(sys.version_info < (3, 12), "Inline Generics not supported before Python3.12")
+    def test_generic_class(self) -> None:
+        exec("class Test[A]: ...")
```
Comment (sobolevn) [inline mypy/test/teststubgen.py:949]: you forgot the `gen.generate_class_stub` part :)

Comment (InvincibleRMC) [reply mypy/test/teststubgen.py:949]: Oh I think I now get what you mean. I was quite confused my apologies.

**Comment by InvincibleRMC** [comment]
@sobolevn any other comments / concerns?

**Comment by InvincibleRMC** [comment]
@sobolevn Friendly ping if you need anything else.

**Pull request closed without merging**